ee/desktop/notify/notify_darwin.go

```diff
@@ -16,9 +18,51 @@ import (
 	"os"
 	"strings"
 	"unsafe"
+
+	"github.com/go-kit/kit/log"
 )
 
-func (d *DesktopNotifier) SendNotification(title, body string) error {
+type macNotifier struct {
+	logger    log.Logger
+	interrupt chan struct{}
+}
+
+func newOsSpecificNotifier(logger log.Logger, _ string) *macNotifier {
+	return &macNotifier{
+		logger:    logger,
+		interrupt: make(chan struct{}),
+	}
+}
+
+func (m *macNotifier) Listen() error {
+	if isBundle() {
+		go func() {
+			for {
+				select {
+				case <-m.interrupt:
+					C.stopNotificationListenerApp()
+					return
+				}
+			}
+		}()
+
+		C.runNotificationListenerApp()
+		return nil
+	} else {
+		for {
+			select {
+			case <-m.interrupt:
+				return nil
+			}
+		}
+	}
+}
```

Comment:
I think this can be simplified to:

```suggestion
func (m *macNotifier) Listen() error {
	if !isBundle() {
		<-m.interrupt
		return nil
	}

	// isBundle
	go func() {
		<-m.interrupt
		C.stopNotificationListenerApp()
		return
	}()

	C.runNotificationListenerApp()
	return nil
}
```

Reply:
Oh yes good catch, thanks! Will update here and in the windows listener